src/Analysis/Ast/Impl/Types/PythonFunctionType.cs

```diff
@@ -141,8 +142,14 @@ public override PythonMemberType MemberType
         internal ImmutableArray<string> Dependencies { get; private set; } = ImmutableArray<string>.Empty;
 
         internal void AddOverload(IPythonFunctionOverload overload) {
-            lock (_lock) {
+            if (_overload != null) {
+                _overloads = new List<IPythonFunctionOverload> { _overload };
```

Comment:
It'd be a bit faster to create the list as `{ _overload, overload }` and return early, rather than doing an `Add` later on the fresh list.

Reply:
Sure